src/core/Form/DateInput/DateInput.tsx

```diff
@@ -0,0 +1,482 @@
+   * @param {FocusEvent<HTMLInputElement>} event FocusEvent
+   */
+  onBlur?: (event: FocusEvent<HTMLInputElement>) => void;
+  /** Label for the input */
+  labelText: ReactNode;
+  /** Hide or show label. Label element is always present, but can be visually hidden.
+   * @default visible
+   */
+  labelMode?: LabelMode;
+  /** Placeholder text for input. Use only as visual aid, not for instructions. */
+  visualPlaceholder?: string;
+  /** Hint text for input. Date input should show hint for formatting date, for example 'Use format D.M.YYYY'. */
+  hintText?: string;
+  /**
+   * 'default' | 'error' | 'success'
+   * @default default
+   */
+  status?: InputStatus;
+  /** Controlled value */
+  value?: string;
+  /** Set components width to 100% */
+  fullWidth?: boolean;
+  /** Text to mark a field optional. Will be wrapped in parentheses and shown after labelText. */
+  optionalText?: string;
+  /** Debounce time in milliseconds for onChange function. No debounce is applied if no value is given. */
+  debounce?: number;
+  /** Tooltip component for the input's label */
+  tooltipComponent?: ReactElement;
+}
+
+const BaseDateInput = (props: DateInputProps & DatePickerProps) => {
+  const {
+    className,
+    labelText,
+    labelMode,
+    onChange: propOnChange,
+    wrapperProps,
+    optionalText,
+    status,
+    statusText,
+    hintText,
+    visualPlaceholder,
+    id,
+    fullWidth,
+    forwardedRef,
+    debounce,
+    statusTextAriaLiveMode = 'assertive',
+    'aria-describedby': ariaDescribedBy,
+    defaultValue,
+    value,
+    datePickerEnabled = false,
+    datePickerTexts = undefined,
+    language = 'fi',
+    dateAdapter = defaultDateAdapter(),
+    shouldDisableDate,
+    minDate = moveYears(firstDayOfMonth(new Date()), -10),
+    maxDate = moveYears(lastDayOfMonth(new Date()), 10),
+    initialDate,
+    tooltipComponent,
+    ...passProps
+  } = props;
+
+  const hintTextId = `${id}-hintText`;
+
+  const statusTextId = `${id}-statusText`;
+
+  const inputRef = useRef<HTMLInputElement>();
+  const openButtonRef = useRef<HTMLButtonElement>(null);
+
+  const [inputValue, setInputValue] = useState<string>('');
+  const [buttonDateLabel, setButtonDateLabel] = useState<string>('');
+  const [inputValueAsDate, setInputValueAsDate] = useState<Date | null>(null);
+  const [calendarVisible, setCalendarVisible] = useState<boolean>(false);
+  const [texts, setTexts] = useState<InternalDatePickerTextProps>(
+    datePickerDefaultTexts.fi,
+  );
+
+  useEffect(() => {
+    if (defaultValue) {
+      setInputValue(String(defaultValue));
+    }
+  }, []);
+
+  useEffect(() => {
+    // Update open button's aria label if input has a valid date
+    const useValue = 'value' in props && value ? value : inputValue;
+    if (useValue) {
+      const date = validateDatePickerValue(useValue);
+      if (date) {
+        setButtonDateLabel(
+          `${texts.selectedDateLabel} ${cellDateAriaLabel(date, texts)}`,
+        );
```

Comment:
The screen reader reads the datepicker button text very fast, all in a row "choose dateSelected date XXXX". 

Adding some separation characters could make it more cohesive? 
```suggestion
        setButtonDateLabel(
          `${texts.selectedDateLabel}: ${cellDateAriaLabel(date, texts)}`,
        );
```

Reply:
Good point! This [ARIA practices datepicker example](https://w3c.github.io/aria-practices/examples/dialog-modal/datepicker-dialog.html) uses comma as a separator, which sounded good when testing with NVDA

Reply:
I think comma would be good here, since the "selected date" is additional description for the date text

Reply:
Updated this with comma